engine/src/cubos/engine/plugins/window.cpp

```diff
@@ -0,0 +1,42 @@
+#include <cubos/engine/plugins/window.hpp>
+#include <cubos/core/settings.hpp>
+
+static void startup(cubos::core::io::Window& window, ShouldQuit& quit, cubos::core::Settings& settings)
+{
+    quit.value = false;
+    window = cubos::core::io::openWindow(
+        settings.getString("window.title", "Cubos"),
+        {
+            settings.getInteger("window.width", 800),
+            settings.getInteger("window.height", 600)
+        }
+    );
+}
+
+static void pollSystem(const cubos::core::io::Window& window, ShouldQuit& quit)
+{
+    window->pollEvents();
+    if(window->shouldClose())
+    {
+        quit.value = true;
+    }
+}
+
+static void swapBuffersSystem(const cubos::core::io::Window& window)
+{
+    window->swapBuffers();
+}
+
+void cubos::engine::plugins::windowPlugin(Cubos& cubos)
+{
+    cubos
+        .addResource<cubos::core::io::Window>()
+
+        .addStartupSystem(startup, "OpenWindow")
+
+        .addSystem(pollSystem, "Poll")
+        .putStageAfter("Main", "Poll")
```

Comment:
I don't think you need to specify 'Main' here, lets just believe that the users of this plugin tie everything together for now. The future dispatcher will solve this problem for us.

Reply:
Fixed